### DISCUSSION TITLE
Error message with command-completion-default-include-p and Magit

### DISCUSSION THREAD

**Discussion by danielsz** [minad/consult#1051] [Q&A]
If the following is set:

`(setq read-extended-command-predicate #'command-completion-default-include-p)`

Then everything is fine until Magit is invoked for the first time. Once Magit has been invoked in my Emacs session, every time I press M-x, the prompt message shows an error:

`read-extended-command-predicate: transient:magit-bisect:--no-checkout: Symbol’s function definition is void: transient--suffix-only
`

The completions are still shown in the mini-buffer, but the prompt line is polluted with the above error message. It looks like this:

```
1/5666 M-x [read-extended-command-predicate: transient:magit-bisect:--no-checkout: Symbol’s function definition is void: transient--suffix-only]
```

**Comment by minad** [comment]
I don't know the actual reason for this problem, but it happens inside `command-completion-default-include-p`, so it must be fixed there (not a Consult or Vertico issue). Or better make sure that these void symbols don't occur in the first place? An installation problem maybe?

**Reply by danielsz** [reply]
Maybe, but I'd rather suspect the transient package. Furthermore, they are conducting [discussions](https://github.com/magit/transient/pull/273) around `command-completion-default-include-p` which I intend to dive in deeper when I have some time. More generally though, it appears that this setting won't narrow down the commands offered by vertico/consult by a lot, since according to this [redditer](https://www.reddit.com/r/emacs/comments/t0xpqh/marking_commands_for_readextendedcommandpredicate/) not many packages have made the effort to comply with the interface (so far). Any thoughts?

Versions:
Magit 3.3.0-7.dda332b, Transient 0.7.0, Git 2.45.1, Emacs 29.4, gnu/linux

**Reply by minad** [reply]
I've seen efforts to add such predicates to more modes, upstream in Emacs core and in packages. My packages also add such predicates. I'd welcome if more packages were updated as well.
